powershell-adapter/Tests/powershellgroup.resource.tests.ps1

```diff
@@ -81,4 +81,42 @@ Describe 'PowerShell adapter resource tests' {
         $res.actualState.result.count | Should -Be 5
         $res.actualState.result| % {$_.Name | Should -Not -BeNullOrEmpty}
     }
+
+    It 'Verify that ClearCache works in PSAdapter' {
+        # generate the cache
+        $null = dsc resource list '*' -a Microsoft.DSC/PowerShell
+        # call the ClearCache operation
+        $scriptPath = Join-Path $PSScriptRoot '..' 'psDscAdapter' 'powershell.resource.ps1'
+        $null = & $scriptPath -Operation ClearCache
+        # verify that PSAdapter does not find the cache
+        dsc -l debug resource list '*' -a Microsoft.DSC/PowerShell 2> $TestDrive/tracing.txt
+        $LASTEXITCODE | Should -Be 0
+        "$TestDrive/tracing.txt" | Should -FileContentMatchExactly 'Cache file not found'
+    }
+
+    It 'Verify that a new PS Cache version results in cache rebuid' {
+        # generate the cache
+        $null = dsc resource list '*' -a Microsoft.DSC/PowerShell
+        # update the version in the cache file
+        $cacheFilePath = if ($IsWindows) {
+            # PS 6+ on Windows
+            Join-Path $env:LocalAppData "dsc\PSAdapterCache.json"
+        } else {
+            # either WinPS or PS 6+ on Linux/Mac
+            if ($PSVersionTable.PSVersion.Major -le 5) {
+                Join-Path $env:LocalAppData "dsc\WindowsPSAdapterCache.json"
+            } else {
+                Join-Path $env:HOME ".dsc" "PSAdapterCache.json"
+            }
+        }
+        $cache = Get-Content -Raw $cacheFilePath | ConvertFrom-Json
+        $cache.CacheSchemaVersion = 0
```

Comment:
would it be better to have the cache keep a timestamp of the last refresh to compare instead of the trace message?

Reply:
When it comes to validation in a test case -  both methods are fine.
If you're talking about actually storing timestamp in the cache file - I'm not sure what value that brings for the product code, since main focus of this functionality is to easily mark and handle breaking changes in the schema;   btw, such timestamp already exists indirectly - LastWriteTime of the cache file.